packages/foundation/src/next/composed.tsx

```diff
@@ -220,16 +232,35 @@ export function composed<
 
     const viewProps = {
       ...componentProps,
+      slotProps: {},
       styles,
-      tokens,
-      _defaultStyles: displayName ? finalStyles : styles
-    } as TViewProps & IDefaultSlotProps<any>;
+      tokens
+    } as TComponentProps & TViewProps;
 
     if (!options.slots) {
       throw new Error(`Component ${options.displayName || (view && view.name) || ''} is missing slot definitions.`);
     }
 
-    const Slots = typeof options.slots === 'function' ? getSlots(viewProps, options.slots(viewProps)) : getSlots(viewProps, options.slots);
+    const Slots = resolveSlots(options.slots, viewProps) as ISlots<Required<TComponentSlots>>;
+
+    // Translate slot default props into their corresponding prop objects and apply styles to them.
+    for (const key in Slots) {
+      if (Slots.hasOwnProperty(key)) {
+        // Translate slot default props into their corresponding prop objects.
+        if (viewProps.hasOwnProperty(key)) {
+          if ((Slots[key] as any).defaultProp) {
+            viewProps.slotProps[key] = translateShorthand((Slots[key] as any).defaultProp, (viewProps as any)[key] as any);
+          } else {
+            viewProps.slotProps[key] = translateShorthand('children', (viewProps as any)[key] as any);
+          }
+        } else {
+          (viewProps.slotProps as any)[key] = {};
+        }
+
+        // Apply corresponding styles to each slot.
+        (viewProps.slotProps as any)[key].className = finalStyles[key];
+      }
+    }
```

Comment:
By no longer calling `getSlots`, the factory seems to be bypassed. Is there still a need for the factory methods at all? 

We may also want to question whether shorthand overloaded with prop objects is still worthwhile. (I think JD mentioned moving slot props elsewhere as an option.)

If we need to keep this I'd like to understand all of the scenarios. I'm hoping this could be cleaned up a bit, either by pulling out, using the factory method, etc.